pandas/tests/series/test_constructors.py

```diff
@@ -831,3 +831,20 @@ def test_constructor_cast_object(self):
         s = Series(date_range('1/1/2000', periods=10), dtype=object)
         exp = Series(date_range('1/1/2000', periods=10))
         tm.assert_series_equal(s, exp)
+
+    def test_overflow_coersion_signed_to_unsigned(self):
+        Series([-1], dtype='uint8')
```

Comment:
To be more specific with regards to what @jreback said, here's an example:
~~~python
with pytest.raises(OverflowError):
   Series([-1], dtype='uint16')
~~~